webapp/src/templates/directives/header.html

```diff
@@ -115,42 +115,20 @@
+          <span class="fa {{tab.icon}}"></span>
+          <span ng-if="tab.name === 'messages'" class="mm-badge-overlay mm-badge-overlay-top" ng-hide="headerCtrl.unreadCount.message <= 0">
             <span class="mm-badge mm-badge-red mm-badge-border" ng-show="headerCtrl.unreadCount.message < 100">{{headerCtrl.unreadCount.message}}</span>
             <span class="mm-badge mm-badge-red mm-badge-border" ng-show="headerCtrl.unreadCount.message >= 100">100+</span>
           </span>
-        </div>
-        <div class="button-label" translate>Messages</div>
-      </a><a ui-sref="tasks.detail" id="tasks-tab" class="tasks-tab" ng-class="{'selected': headerCtrl.currentTab === 'tasks'}" mm-auth="can_view_tasks,can_view_tasks_tab">
-        <div class="mm-icon" ng-class="{'mm-icon-inverse': headerCtrl.currentTab !== 'tasks'}">
-          <span class="fa fa-flag"></span>
-        </div>
-        <div class="button-label" translate>Tasks</div>
-      </a><a ui-sref="reports.detail" id="reports-tab" class="reports-tab" ng-class="{'selected': headerCtrl.currentTab === 'reports'}" mm-auth="can_view_reports,can_view_reports_tab">
-        <div class="mm-icon" ng-class="{'mm-icon-inverse': headerCtrl.currentTab !== 'reports'}">
-          <span class="fa fa-list-alt"></span>
-          <span class="mm-badge-overlay mm-badge-overlay-top" ng-hide="headerCtrl.unreadCount.report <= 0">
+          <span ng-if="tab.name === 'reports'" class="mm-badge-overlay mm-badge-overlay-top" ng-hide="{{headerCtrl.unreadCount.report <= 0}}">
             <span class="mm-badge mm-badge-red mm-badge-border" ng-show="headerCtrl.unreadCount.report < 100">{{headerCtrl.unreadCount.report}}</span>
             <span class="mm-badge mm-badge-red mm-badge-border" ng-show="headerCtrl.unreadCount.report >= 100">100+</span>
           </span>
         </div>
-        <div class="button-label" translate>Reports</div>
-      </a><a ui-sref="contacts.detail" id="contacts-tab" class="contacts-tab" ng-class="{'selected': headerCtrl.currentTab === 'contacts'}" mm-auth="can_view_contacts,can_view_contacts_tab">
-        <div class="mm-icon" ng-class="{'mm-icon-inverse': headerCtrl.currentTab !== 'contacts'}">
-          <span class="fa fa-user"></span>
-        </div>
-        <div class="button-label" translate>Contacts</div>
-      </a><a ui-sref="analytics" ui-sref-opts="{reload: true}" id="analytics-tab" class="analytics-tab" ng-class="{'selected': headerCtrl.currentTab === 'analytics'}" mm-auth="can_view_analytics,can_view_analytics_tab">
-        <div class="mm-icon" ng-class="{'mm-icon-inverse': headerCtrl.currentTab !== 'analytics'}">
-          <span class="fa fa-bar-chart-o"></span>
-        </div>
-        <div class="button-label" translate>Analytics</div>
+        <div ng-if="headerCtrl.isMobile() && headerCtrl.filteredTabs().length <= 3 || !headerCtrl.isMobile()" translate>{{tab.translation}}</div>
```

Comment:
Is the `headerCtrl.isMobile()` addition in your boolean expression adding value here? Think about the different cases. This can be simplified.